### PR TITLE
optimize/flag: flag params are prepended to init or main

### PR DESCRIPTION
**Pull request by 1046102779**
panic msg: flag redefined: kubeconfig

### CODE REVIEW

**Comment by 1046102779** [comment]
#3917

File: utils/utils.go [hunk 1]
```diff
@@ -111,3 +102,14 @@ func StringSliceContains(needle string, haystack []string) bool {
 	}
 	return false
 }
+
+// SetEnvVariables set variables to environment.
+func SetEnvVariables(variables map[string]string) error {
+	for key, value := range variables {
+		err := os.Setenv(key, value)
+		if err != nil {
+			return err
```
Comment (daixiang0) [inline utils/utils.go:111]: Please add UT.

File: utils/utils_test.go [hunk 2]
```diff
@@ -111,3 +112,13 @@ func TestStringSliceContains(t *testing.T) {
 		assert.False(t, StringSliceContains("not-in-item", nil))
 	})
 }
+
+func TestSetEnvVariables(t *testing.T) {
```
Comment (daixiang0) [inline utils/utils_test.go:116]: Add false case please.

File: pkg/runtime/cli_test.go [hunk 3]
```diff
@@ -46,7 +48,7 @@ func TestSetEnvVariables(t *testing.T) {
 			"ABC_HOST": "456",
 		}
 
-		err := setEnvVariables(variables)
+		err := utils.SetEnvVariables(variables)
```
Comment (daixiang0) [inline pkg/runtime/cli_test.go:51]: Still need false case here.

File: utils/utils.go [hunk 4]
```diff
@@ -41,18 +40,10 @@ func GetConfig() *rest.Config {
...
-
-	var kubeconfig *string
-	if home := homedir.HomeDir(); home != "" {
-		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
-	} else {
-		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
-	}
-	flag.Parse()
-
+	kubeconfig := os.Getenv(KubeConfigVar)
```
Comment (daixiang0) [inline utils/utils.go:43]: Need to check empty? Please add UT here.

File: pkg/runtime/cli_test.go (hunk 3 above)
Comment (1046102779) [reply pkg/runtime/cli_test.go]: this UT is invalid, i deleted it.

**Comment by yaron2** [comment]
@1046102779  see tests are failing.

**Comment by 1046102779** [comment]
All  this kind of error:

```shell
error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
error: 3 bytes of length header were received
fatal: expected flush after ref listing
```

I don't know how to solve it.

**Pull request closed without merging**